src/erlang_ls_tree.erl

```diff
@@ -198,7 +198,9 @@ attribute(Tree, Extra) ->
     {spec, {spec, {{F, A}, _}}} ->
       SpecLocations = maps:get(spec_locations, Extra, []),
       Locations     = proplists:get_value({F, A}, SpecLocations),
-      [erlang_ls_poi:poi(Tree, type_application, {T, L}, Extra) || {T, L} <- Locations];
+      [ erlang_ls_poi:poi(Tree, spec, {{F, A}, Tree}, Extra)
```

Comment:
When encountering specs, we were only indexing the types and not the spec itself. This is not the case anymore. The reason for passing the whole `Tree` as the third argument to the `poi/4` is so that we can then use the convenient `erl_prettypr` module to pretty-print the spec. We could have passed the pretty printed version but that would have been worse performance-wise. Also, it could turn useful to get the whole raw attribute in the DB.